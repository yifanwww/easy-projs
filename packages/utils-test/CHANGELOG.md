# CHANGELOG
## @packages/utils-test v0.2.0 (Unreleased)
### Features

- Add `mockLocalStorage` and `mockSessionStorage` to help mock local storage and session storage
- Add test utils: `expectElementSnapshot`, `expectSnapshot`, `mockConsoleToMatchSnapshot`

## @packages/utils-test v0.1.0 (2021-10-29)
### Features

- Add `test.setup.ts` to setup jest globally for every package
- Add `validateHookValueNotChanged` to help test React hooks
