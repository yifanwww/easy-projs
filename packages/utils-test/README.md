# @easy/utils-test

Test utils used for easy-projs.

## Develop this package

You need to build package `@easy/scripts` before building or testing this package.

### Build this package

Execute `pnpm run build` to build this package.
