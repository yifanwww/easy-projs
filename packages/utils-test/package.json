{
    "name": "@easy-pkg/utils-test",
    "version": "0.1.0",
    "description": "Test utils used for easy-projs.",
    "main": "lib/index.js",
    "module": "lib/index.js",
    "typings": "lib/index.d.ts",
    "homepage": "https://github.com/yifanwww/easy-projs#readme",
    "license": "MIT",
    "author": "yifanwww <yifanw1101@gmail.com> (https://github.com/yifanwww)",
    "bugs": {
        "url": "https://github.com/yifanwww/easy-projs/issues"
    },
    "private": true,
    "repository": {
        "type": "git",
        "url": "git+https://github.com/yifanwww/easy-projs.git",
        "directory": "packages/utils-test"
    },
    "scripts": {
        "build": "concurrently -n esm,types \"pnpm run build:esm\" \"pnpm run build:types\"",
        "build:esm": "tsc --project tsconfigs/tsconfig.esm.json",
        "build:types": "tsc --project tsconfigs/tsconfig.types.json",
        "clean": "rimraf lib tsconfigs/*.tsbuildinfo",
        "dev": "pnpm run build:esm --watch",
        "format": "prettier --write \"**/*.{cjs,css,html,js,jsx,mjs,json,less,scss,ts,tsx,yaml,yml}\"",
        "lint": "eslint .",
        "pre-commit": "lint-staged",
        "test": "unit-test",
        "test-full": "unit-test",
        "typecheck": "tsc --project tsconfig.json --noEmit"
    },
    "dependencies": {
        "@testing-library/react-hooks": "^8.0.1",
        "react": "^17.0.2",
        "react-dom": "^17.0.2"
    },
    "devDependencies": {
        "@easy-config/eslint-config": "workspace:*",
        "@easy-config/global-types": "workspace:*",
        "@easy-config/scripts": "workspace:*",
        "@easy-config/tsconfigs": "workspace:*",
        "@testing-library/jest-dom": "^5.16.5",
        "concurrently": "^7.4.0",
        "eslint": "^8.24.0",
        "lint-staged": "^13.0.3",
        "prettier": "2.7.1",
        "rimraf": "^3.0.2",
        "type-fest": "^3.2.0",
        "typescript": "4.8.3"
    }
}
