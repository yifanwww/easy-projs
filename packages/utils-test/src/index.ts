export * from './console';
export * from './ReactHook';
export * from './storage';
