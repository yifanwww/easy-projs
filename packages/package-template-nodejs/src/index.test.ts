import { expectSnapshot } from '@easy/utils-test';

import * as package_exports from './index';

it('exports from this package', () => expectSnapshot(Object.keys(package_exports)));

it('Test function `templateNodejs`', () => expectSnapshot(package_exports.templateNodejs()));
