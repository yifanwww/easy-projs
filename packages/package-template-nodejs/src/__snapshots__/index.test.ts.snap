// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test function \`templateNodejs\` 1`] = `"This package named \`@easy/package-template-nodejs\` is used for creating new nodejs packages."`;

exports[`exports from this package 1`] = `
Array [
  "templateNodejs",
]
`;
